test/modules/TestResultCoordinator/TestReportUtil.cs

```diff
@@ -103,9 +110,100 @@ internal static List<ITestReportMetadata> ParseReportMetadataJson(string reportM
             return reportMetadataList;
         }
 
+        internal static async Task<Uri> GetOrCreateBlobContainerSasUriForLogAsync(string storageAccountConnectionString)
+        {
+            string containerName = GetAzureBlobContainerNameForLog();
+            var containerClient = new BlobContainerClient(storageAccountConnectionString, containerName);
+
+            if (!await containerClient.ExistsAsync())
+            {
+                await containerClient.CreateAsync();
+            }
+
+            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(storageAccountConnectionString);
+            var container = new CloudBlobContainer(containerClient.Uri, storageAccount.Credentials);
+            return GetContainerSasUri(container);
+        }
+
+        internal static async Task UploadLogsAsync(string iotHubConnectionString, Uri blobContainerWriteUri, ILogger logger)
+        {
+            Preconditions.CheckNonWhiteSpace(iotHubConnectionString, nameof(iotHubConnectionString));
+            Preconditions.CheckNotNull(blobContainerWriteUri, nameof(blobContainerWriteUri));
+            Preconditions.CheckNotNull(logger, nameof(logger));
+
+            DateTime uploadLogStartAt = DateTime.UtcNow;
+            logger.LogInformation("Send upload logs request to edgeAgent.");
+
+            ModuleClient moduleClient = await ModuleClient.CreateFromEnvironmentAsync(TransportType.Amqp_Tcp_Only);
```

Comment:
This has to be AMQP? Shouldn't it be whatever protocol we're using to send upstream?

Reply:
This is used for module client to connect to edgeHub and routing to edgeAgent on IoT Edge device.  And this is not related to the test; therefore I chose to use AMQP for this request.